middle_end/flambda/inlining_cost.ml

```diff
@@ -89,7 +89,12 @@ let lambda_smaller' lam ~than:threshold =
       List.iter (fun (_, lam) -> lambda_named_size lam) bindings;
       lambda_size body
     | Switch (_, sw) ->
-      let aux = function _::_::_ -> size := !size + 5 | _ -> () in
+      let rec aux =
+        function
+        | [] | [_] -> ()
+        | [_;_] -> size := !size + 5
+        | _ :: t -> size := !size + 1; aux t
+      in
       aux sw.consts; aux sw.blocks;
```

Comment:
I think the code would be easier to read as

```ocaml
let cost cases =
 let size = List.length cases in
 if size <= 1 then 0
 else 3 + size
in
size := !size + cost sw.consts + cost sw.blocks;
```

(It's a bit odd that a switch with one const and one block case has cost 0, but if I understand correctly you don't really care?)

Reply:
Yes it might be worth taking into account the size for that 'if'. The thing is that the parameters for inlining have been carefully tuned through lots of benchmarking. Changing the computation too much would requires retunning everything.
So it would better reflect the actual inlining cost. But this is too much effort for this.

Reply:
I shamelessly took your suggestion. What should I put in the change since there is no code from me ?

Reply:
Well sometimes we use "Foo and bar, review by blah", but "bar" and "blah" have to be distinct in this case, so I would recommend keeping me as a reviewer instead ;-)

Reply:
> The thing is that the parameters for inlining have been carefully tuned through lots of benchmarking. Changing the computation too much would requires retuning everything.

At first I thought this justification made perfect sense, but: don't you also need (re)tuning if you change the cost of higher-than-2 matches? I thought this was a case of "oh we don't need to benchmark this change because it does not matter this strongly", but your comment seems to imply that maybe we would need to.

Reply:
(Well one argument would be that the change you propose does not affect pattern-matching on list, or any of the common datastructures, while the change for one-const-and-one-block would.)